pkg/controller/periodic/periodic_test.go

```
@@ -1968,9 +1969,26 @@ func TestGetDataGatherCR(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			insightsClientset := insightsFakeCli.NewClientset(tt.dataGather)
-			kubeClientset := kubefake.NewClientset(tt.deployment)
-			mockController := NewWithTechPreview(nil, nil, nil, nil, kubeClientset, insightsClientset.InsightsV1(), nil, nil, nil, nil)
+			insightsClientset := insightsFakeCli.NewSimpleClientset(tt.dataGather)
```

golangci-lint (2.12.2)

[error] SA1019: insightsFakeCli.NewSimpleClientset is deprecated: NewClientset replaces this with support for field management, which significantly improves server side apply testing. NewClientset is only available when apply configurations are generated (e.g. via --with-applyconfig). 

(staticcheck)